api/src/main/java/org/openmrs/module/queue/model/Queue.java

```diff
@@ -52,13 +55,59 @@ public class Queue extends BaseChangeableOpenmrsMetadata {
 	private Concept service;
 	
 	@OneToMany(mappedBy = "queue", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
-	@Where(clause = "voided = 0 and (started_at <= current_timestamp() and ended_at is null)")
 	private List<QueueEntry> queueEntries;
 	
 	@OneToMany(mappedBy = "queue", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
-	@Where(clause = "retired = 0")
 	private List<QueueRoom> queueRooms;
 	
+	/**
+	 * @return all non-voided QueueEntries that do not start in the future and are not already ended
+	 */
+	public List<QueueEntry> getActiveQueueEntries() {
+		List<QueueEntry> l = new ArrayList<>();
+		Date now = new Date();
+		if (queueEntries != null) {
+			for (QueueEntry e : queueEntries) {
+				if (BooleanUtils.isNotTrue(e.getVoided())) {
+					if (!e.getStartedAt().after(now) && e.getEndedAt() == null) {
```

Comment:
Note:  the intention here was to exactly replicate the pre-existing logic.  I don't know if this is what we want or not (eg. we consider the possibility of "future" queue entries, and don't want entries that _start_ in the future to be considered active, but we don't consider the possibility of queue entries that have already started but whose endDate is in the future, but rather anything with an end date is inactive.  We can discuss the correctness of this logic, and happy to change this to whatever we come up with, but for the initial PR, just replicating existing behavior.

Reply:
Yeah, there's maybe some discussion to be had. I'm not exactly sure what the concrete use-case for future-dated entries is, but it seems like it's getting out of the realm of a "list of patients waiting for a resource".